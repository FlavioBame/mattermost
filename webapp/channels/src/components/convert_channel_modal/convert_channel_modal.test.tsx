// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Modal} from 'react-bootstrap';

import {General} from 'mattermost-redux/constants';

import ConvertChannelModal from 'components/convert_channel_modal/convert_channel_modal';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

describe('component/ConvertChannelModal', () => {
    const updateChannelPrivacy = jest.fn().mockImplementation(() => Promise.resolve({}));
    const baseProps = {
        onExited: jest.fn(),
        channelId: 'owsyt8n43jfxjpzh9np93mx1wa',
        channelDisplayName: 'Channel Display Name',
        actions: {
            updateChannelPrivacy,
        },
    };

    test('should match snapshot for convert_channel_modal', () => {
        const wrapper = shallow(
            <ConvertChannelModal {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match state when onHide is called', () => {
        const wrapper = shallow<ConvertChannelModal>(
            <ConvertChannelModal {...baseProps}/>,
        );

        wrapper.setState({show: true});
        wrapper.instance().onHide();
        expect(wrapper.state('show')).toEqual(false);
    });

    test('should call updateChannelPrivacy after confirming', (done) => {
        baseProps.onExited.mockImplementation(() => done());

        const wrapper = mountWithIntl(
            <ConvertChannelModal
                {...baseProps}
            />,
        );

        expect(wrapper.find(Modal).prop('show')).toBe(true);
        expect(baseProps.onExited).not.toHaveBeenCalled();

        wrapper.find('[data-testid="convertChannelConfirm"]').simulate('click');

        expect(updateChannelPrivacy).toHaveBeenCalledTimes(1);
        expect(updateChannelPrivacy).toHaveBeenCalledWith(baseProps.channelId, General.PRIVATE_CHANNEL);
        expect(wrapper.find(Modal).prop('show')).toBe(false);
    }, 5000);

    test('should call onExited after cancelling', (done) => {
        baseProps.onExited.mockImplementation(() => done());

        const wrapper = mountWithIntl(
            <ConvertChannelModal
                {...baseProps}
            />,
        );

        expect(wrapper.find(Modal).prop('show')).toBe(true);
        expect(baseProps.onExited).not.toHaveBeenCalled();

        wrapper.find('[data-testid="convertChannelCancel"]').simulate('click');

        expect(wrapper.find(Modal).prop('show')).toBe(false);
    }, 5000);
});
