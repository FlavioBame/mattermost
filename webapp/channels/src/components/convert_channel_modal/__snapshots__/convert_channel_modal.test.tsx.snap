// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/ConvertChannelModal should match snapshot for convert_channel_modal 1`] = `
<Modal
  animation={true}
  aria-labelledby="convertChannelModalLabel"
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogClassName="a11y__modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onExited={[MockFunction]}
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  role="dialog"
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h1"
      id="convertChannelModalLabel"
    >
      <MemoizedFormattedMessage
        defaultMessage="Convert {display_name} to a Private Channel?"
        id="convert_channel.title"
        values={
          Object {
            "display_name": "Channel Display Name",
          }
        }
      />
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <p>
      <MemoizedFormattedMessage
        defaultMessage="When you convert <b>{display_name}</b> to a private channel, history and membership are preserved. Publicly shared files remain accessible to anyone with the link. Membership in a private channel is by invitation only."
        id="convertChannel.question1"
        values={
          Object {
            "b": [Function],
            "display_name": "Channel Display Name",
          }
        }
      />
    </p>
    <p>
      <MemoizedFormattedMessage
        defaultMessage="Are you sure you want to convert <b>{display_name}</b> to a private channel?"
        id="convertChannel.question3"
        values={
          Object {
            "b": [Function],
            "display_name": "Channel Display Name",
          }
        }
      />
    </p>
  </ModalBody>
  <ModalFooter
    bsClass="modal-footer"
    componentClass="div"
  >
    <button
      className="btn btn-tertiary"
      data-testid="convertChannelCancel"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="No, cancel"
        id="convert_channel.cancel"
      />
    </button>
    <button
      autoFocus={true}
      className="btn btn-primary"
      data-dismiss="modal"
      data-testid="convertChannelConfirm"
      onClick={[Function]}
      type="button"
    >
      <MemoizedFormattedMessage
        defaultMessage="Yes, convert to private channel"
        id="convert_channel.confirm"
      />
    </button>
  </ModalFooter>
</Modal>
`;
